materialize-bigquery/transactor.go

```diff
@@ -0,0 +1,280 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/bigquery"
+	pm "github.com/estuary/protocols/materialize"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+type transactor struct {
+	ctx      context.Context
+	ep       *Endpoint
+	fence    *fence
+	bindings []*binding
+}
+
+func (t *transactor) Load(it *pm.LoadIterator, _ <-chan struct{}, loaded func(int, json.RawMessage) error) error {
+
+	var err error
+
+	// Iterate through all of the Load requests being made.
+	for it.Next() {
+
+		var b = t.bindings[it.Binding]
+
+		// Check to see if we have a keyFile open already in gcs for this binding and if not, create one.
+		if b.load.keysFile == nil {
+			b.load.keysFile, err = t.ep.NewExternalDataConnectionFile(
```

Comment:
Seems this was already discussed. I guess most use cases for big query has very large transaction, right? It will benefit most from the file creation strategy.